packages/orm/src/Promise.ts

```diff
@@ -0,0 +1,38 @@
+import { Entity, Loaded, LoadHint } from "./EntityManager";
+import { getEm } from "./index";
+
+export {};
+
+declare global {
+  interface Promise<T> {
```

Comment:
I appreciate the creativity here :-) but yeah, I'm not comfortable with this level of prototype hacking, both a) outside of `graphql-service`, and b) these `populate` methods only allow to promises of entities and not all promises.

Reply:
was thinking maybe we just provide the method definitions  in their own interface and a helper function that would add the implementation to `Promise.prototype`.  Then consumers could apply it globally on an opt in basis.

eg we'd define something like this:

```typescript
export interface PromisePopulate<T> {
  populate<T extends Entity, H extends LoadHint<T>>(this: Promise<T>, hint: H): Promise<Loaded<T, H>>;
  ...
}
export function addPopulateToPromise() { ... }
```

and then codebases that use joist could add this somewhere to get the behavior:

```typescript
declare global { interface Promise<T> extends PromisePopulate<T> {} }
addPopulateToPromise();
```